Client1/Client.py

```diff
@@ -0,0 +1,164 @@
+import requests
+from Model.Player import Player
+from Model.Room import Room
+from Common.AuthenticationErrors import AuthenticationErrors
+from Common.RoomsAPIErrors import RoomsAPIErrors
+
+
+class Client:
+    player_token = None
+
+    @classmethod
+    def main(cls):
+        cls.display_authentication_menu()
+
+    @classmethod
+    def display_authentication_menu(cls):
+        auth_menu = f"Welcome to Joker Game!\n" \
+                    f"Please choose one of the following options:\n" \
+                    f"1. Login\n" \
+                    f"2. Register\n"
+        print(auth_menu)
+        player_choice = input()
+        if player_choice == '1':
+            cls.login()
+        elif player_choice == '2':
+            cls.register()
+
+    @classmethod
+    def login(cls):
+        print("Welcome to Login page!\n")
+        username = input("Please insert your username: ")
+        password = input("Please insert your password: ")
+        print("")
+        data = {'username': username, 'password': password}
+        response = requests.post("http://127.0.0.1:5000/login", json=data)
+        if (response.text == AuthenticationErrors.MISSING_LOGIN_DETAILS.value) or \
+           (response.text == AuthenticationErrors.USERNAME_DOESNT_EXIST.value) or \
+           (response.text == AuthenticationErrors.WRONG_PASSWORD.value):
+            print(response.text)
+            print("")
+            cls.login()
+        else:
+            response_data = response.json()
+            cls.player_token = response_data['token']
+            print("You have successfully logged in\n")
+            cls.display_main_menu()
+
+    @classmethod
+    def register(cls):
+        print("Welcome to Registration page!\n")
+        username = input("Please insert a username: ")
+        password = input("Please insert a password: ")
+        print("")
+        data = {'username': username, 'password': password}
+        response = requests.post("http://127.0.0.1:5000/register", json=data)
+        print(response.text)
```

Comment:
the register shouldn't return a message to the client, it can return the user or error message, or boolean if success.